File: js/task2b/S2_FOR_03.js
// S2_FOR_03 - Sum until threshold

function sumUntil(nums, threshold) {
  let sum = 0;
  for (const n of nums) {
    if (sum + n > threshold) {
      break;
    }
    sum += n;
  }
  return sum;
}

// tests
console.log(sumUntil([1, 2, 3, 4, 5], 10)); // 10 (1+2+3+4)
console.log(sumUntil([1, 2, 3, 4, 5], 6));  // 6 (1+2+3)
console.log(sumUntil([5, 5, 5], 7));         // 5
console.log(sumUntil([10, 20], 5));          // 0
console.log(sumUntil([], 100));              // 0

File: python/task2b/S2_FOR_03.py
# S2_FOR_03 - Sum until threshold

def sum_until(nums, threshold):
    total = 0
    for n in nums:
        if total + n > threshold:
            break
        total += n
    return total

# Tests
print(f"sum_until([1,2,3,4,5], 6): {sum_until([1,2,3,4,5], 6)}")
print(f"sum_until([1,2,3,4,5], 100): {sum_until([1,2,3,4,5], 100)}")
print(f"sum_until([10,20,30], 15): {sum_until([10,20,30], 15)}")
print(f"sum_until([], 10): {sum_until([], 10)}")
